core/src/main/java/org/apache/iceberg/ManifestFiles.java

```diff
@@ -141,7 +141,23 @@ public static ManifestReader<DataFile> read(
    * @return a manifest writer
    */
   public static ManifestWriter<DataFile> write(PartitionSpec spec, OutputFile outputFile) {
-    return write(1, spec, outputFile, null);
+    return write(spec, outputFile, ImmutableMap.of());
+  }
+
+  /**
+   * Create a new {@link ManifestWriter}.
+   *
+   * <p>Manifests created by this writer have all entry snapshot IDs set to null. All entries will
+   * inherit the snapshot ID that will be assigned to the manifest on commit.
+   *
+   * @param spec {@link PartitionSpec} used to produce {@link DataFile} partition tuples
+   * @param outputFile the destination file location
+   * @param config the configuration used by the Manifest writer
+   * @return a manifest writer
+   */
+  public static ManifestWriter<DataFile> write(
+      PartitionSpec spec, OutputFile outputFile, Map<String, String> config) {
```

Comment:
These methods should always include the format version.

Also, I don't think it is a good idea to add a property map throughout.

Reply:
The write method without format version is already referenced in the code base prior to my changes.

https://github.com/apache/iceberg/blob/3170e6ebc65b0c126e57956fd2df5efc59ff88ed/core/src/main/java/org/apache/iceberg/ManifestFiles.java#L143

Thus, I copied the signature of the above `write` method and added `compressionCodec` and `compressionLevel` as additional parameters.

Reply:
Furthermore, if the caller decides to use the `AVRO_COMPRESSION_LEVEL_DEFAULT` value then using `int` as the datatype would not be possible.

https://github.com/apache/iceberg/blob/31edba2a5f49b9e948c5f50981f1cc47bc94780c/core/src/main/java/org/apache/iceberg/TableProperties.java#L177